Neos.Flow/Classes/Http/Cookie.php

```diff
@@ -122,7 +122,7 @@ public function __construct($name, $value = null, $expires = 0, $maximumAge = nu
         }
 
         $this->name = $name;
-        $this->value = $value;
+        $this->value = $value === null ? null : preg_replace('/^"(.*)"$/', '$1', $value);
```

Comment:
I would prefer another constraint check like the other ones above that throws an `\InvalidArguments` exception if the value does not adhere to the rules and then fix the code that _creates_ the instance.
That way the risk of unplanned side effects is limited.

Also, this should be covered with a test :)

Reply:
Ok but the issue is that the quotes aren't added by Flow but by the Client.
So basically Flow is setting the value without quotes and on the next requests, the client adds quotes around it.
Yep, tests should be there - is there a documentation on how to setup and run the tests for Flow? I never did this. We're using Codeception in our stuff.

Reply:
> Ok but the issue is that the quotes aren't added by Flow but by the Client.

But Flow calls the constructor, doesn't it?
I think that part should be using [Cookie::createFromRawSetCookieHeader()](https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Classes/Http/Cookie.php#L172) (or a new static constructor if that doesn't fit) which should take care of the trimming.

> is there a documentation on how to setup and run the tests for Flow?

Not really..
You can have a look at the existing tests to get an idea.
Executing unit tests is as simple as running:
```
./bin/phpunit --colors -c Build/BuildEssentials/PhpUnit/UnitTests.xml
```

But don't worry, I can help out with the tests!

Reply:
Note: As of 6.3 **development-distribution** you can also just run `./composer test-unit`, `./composer test-functional` and/or `./composer test-static` inside the flow root directory
See https://github.com/neos/flow-development-distribution/blob/master/composer.json#L40-L42